components/shopify/actions/create-smart-collection/create-smart-collection.mjs

```diff
@@ -0,0 +1,97 @@
+import shopify from "../../shopify.app.mjs";
+import rules from "../common/rules.mjs";
+import { toSingleLineString } from "../commons.mjs";
+
+export default {
+  key: "shopify-create-smart-collection",
+  name: "Create Smart Collection",
+  description: toSingleLineString(`
+    Creates a smart collection.
+    You can fill in any number of rules by selecting more than one option in each prop.
+    [See docs](https://shopify.dev/api/admin-rest/2021-10/resources/smartcollection#post-smart-collections)
+  `),
+  version: "0.0.1",
+  type: "action",
+  props: {
+    shopify,
+    title: {
+      propDefinition: [
+        shopify,
+        "title",
+      ],
+      description: "Title of the smart collection",
+    },
+    disjunctive: {
+      type: "boolean",
+      label: "Disjunctive",
+      description: toSingleLineString(`
+        If \`false\`, the product must match all the rules to be included in the smart collection.
+        Otherwise, it only needs to match at least one rule
+      `),
+      optional: true,
+      default: false,
+    },
+    rules: {
+      type: "integer",
+      label: "Number of rules",
+      description: "The number of rules to input",
+      default: 1,
+      min: 1,
+      reloadProps: true,
+    },
+  },
+  async additionalProps() {
```

Comment:
-OPTIONAL-
I think we can use `reduce` and return it directly here. Sounds more elegant to me. But completely optional :)